frontend/components/markup-modal/markup-modal-content.js

```diff
@@ -0,0 +1,204 @@
+    baseUrl: PropTypes.string.isRequired,
+  }
+
+  indicatorRef = React.createRef()
+
+  state = {
+    path: '',
+    link: '',
+    markupFormat: 'link',
+    message: undefined,
+  }
+
+  get baseUrl() {
+    const { baseUrl } = this.props
+    if (baseUrl) {
+      return baseUrl
+    } else {
+      // Default to the current hostname for when there is no `BASE_URL` set
+      // at build time (as in most PaaS deploys).
+      const { protocol, hostname } = window.location
+      return `${protocol}//${hostname}`
+    }
+  }
+
+  generateBuiltBadgeUrl() {
+    const { baseUrl } = this
+    const { path, queryString } = this.state
+
+    const suffix = queryString ? `?${queryString}` : ''
+    return `${baseUrl}${path}.svg${suffix}`
+  }
+
+  renderLivePreview() {
+    // There are some usability issues here. It would be better if the message
+    // changed from a validation error to a loading message once the
+    // parameters were filled in, and also switched back to loading when the
+    // parameters changed.
+    const { baseUrl } = this.props
+    const { pathIsComplete } = this.state
+    let src
+    if (pathIsComplete) {
+      src = this.generateBuiltBadgeUrl()
+    } else {
+      src = staticBadgeUrl(
+        baseUrl,
+        'preview',
+        'some parameters missing',
+        'lightgray'
+      )
+    }
+    return (
+      <p>
+        <Badge display="block" src={src} />
+      </p>
+    )
+  }
+
+  copyMarkup = async markupFormat => {
+    const {
+      example: {
+        example: { title },
+      },
+    } = this.props
+    const { link } = this.state
+
+    const builtBadgeUrl = this.generateBuiltBadgeUrl()
+    const markup = generateMarkup({
+      badgeUrl: builtBadgeUrl,
+      link,
+      title,
+      markupFormat,
+    })
+
+    try {
+      await clipboardCopy(markup)
+    } catch (e) {
+      this.setState({
+        message: 'Copy failed',
+        markup,
+      })
+      return
+    }
+
+    this.setState({ markup })
+    this.indicatorRef.current.trigger()
+  }
+
+  renderMarkupAndLivePreview() {
+    const { indicatorRef } = this
+    const { markup, message, pathIsComplete } = this.state
+
+    return (
+      <div>
+        {this.renderLivePreview()}
+        <CopiedContentIndicator
+          ref={indicatorRef}
+          copiedContent={<span style={{ width: '100px' }}>{markup}</span>}
+        >
+          <RequestMarkupButtom
+            isDisabled={!pathIsComplete}
+            onMarkupRequested={this.copyMarkup}
+          />
+        </CopiedContentIndicator>
+        {message && (
+          <div>
+            <p>{message}</p>
+            <p>Markup: {markup}</p>
+          </div>
+        )}
+      </div>
+    )
+  }
+
+  renderDocumentation() {
+    const {
+      example: { documentation },
+    } = this.props
+
+    return documentation ? (
+      <Documentation dangerouslySetInnerHTML={{ __html: documentation }} />
+    ) : null
+  }
+
+  // renderFullPattern() {
```

Comment:
I think it would be useful to show the full pattern and/or the full example somewhere in the modal. I worked on that a bit but ended up deferring it. This is some of the remnant code; not sure whether it is worth keeping.